Realm/Realm/Native/PrimitiveValue.cs

```diff
@@ -271,5 +274,25 @@ private struct LinkValue
         {
             public IntPtr object_ptr;
         }
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct TimestampValue
+        {
+            private const long UnixEpochTicks = 621355968000000000;
+            private const long TicksPerSecond = 10000000;
+            private const long NanosecondsPerTick = 100;
+
+            private long seconds;
```

Comment:
Shouldn't these two be public?

Reply:
This struct is only used for marshaling - private members are correctly marshaled by the runtime and there's a constructor that accepts ticks that sets `seconds` and `nanoseconds` to the correct value based on the conversion formula.

So from a consumer's point of view, they should only care about the `ToTicks` and `FromTicks` (a.k.a. constructor) API.

Reply:
Makes sense. Maybe it'll make sense to still make those public to make it consistent with `StringValue`, `BinaryValue` and `LinkValue`

Reply:
I would rather make the fields in `StringValue` and friends private and encapsulate any relevant logic in those structs, but will do that in a separate PR.